realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -854,16 +870,10 @@ public static void insertOrUpdate(Realm realm, Iterator<? extends RealmModel> ob
                 if (rowIndex == TableOrView.NO_MATCH) {
                     rowIndex = Table.nativeAddEmptyRow(tableNativePtr, 1);
                     if (primaryKeyValue != null) {
-                        Table.nativeSetString(tableNativePtr, pkColumnIndex, rowIndex, ((AllTypesRealmProxyInterface) object).realmGet$columnString());
+                        Table.nativeSetString(tableNativePtr, pkColumnIndex, rowIndex, (String)primaryKeyValue);
```

Comment:
Yup. The default is based on the `@Required` annotation.

I am actually not entirely sure we handle that case properly here: `@Required @PrimaryKey String  id`, should use `'"` as the default value.